proof_of_concept/definitions/identifier.py

```diff
@@ -44,7 +44,7 @@ def __new__(cls: Type['Identifier'], seq: Any) -> 'Identifier':
                 if not cls._segment_regex.match(segment):
                     raise ValueError(f'Invalid identifier segment {segment}')
 
-        return str.__new__(cls, seq)        # type: ignore
+        return str.__new__(cls, seq)
```

Comment:
does not need to be ignored? ;-)

Reply:
Ah yes, welcome to mypy. This is the correct way to create a class derived from str (I think, it's not very extensively documented), but the problem is that this statement returns an object of type `str`, while the signature wants an `Identifier`, so mypy 0.760 gave an error. I'm not sure actually why I used `# type ignore` instead of a cast, but that disabled the error.

Recently, mypy 0.800 was released, and apparently it now knows about deriving from `str` making the `# type: ignore` superfluous. And since superfluous `# type ignore` statements are an error in mypy, I removed it so the tests would pass...